src/LuaTransformer.ts

```diff
@@ -3072,6 +3078,10 @@ export class LuaTransformer {
     }
 
     public transformParenthesizedExpression(expression: ts.ParenthesizedExpression): tstl.Expression {
+        if (ts.isAssertionExpression(expression.expression)) {
```

Comment:
What about `ts.isAsExpression`

Reply:
`AssertionExpression` is defined as `type AssertionExpression = TypeAssertion | AsExpression;` in TS